### PR TITLE
Fe customer bookings

### PR DESCRIPTION
**Pull request by wiroger9595**
### What this PR does (required):
- create customer booking UI page and mobile page 

### Screenshots / Videos (front-end only):

Screen Shot 2022-02-16 at 1 21 39 PM

Screen Shot 2022-02-16 at 1 22 01 PM

### Any information needed to test this feature (required):
- it need integrate with stripe API so it can proceed with the payment

### CODE REVIEW

File: client/src/components/TabPanel/TabPanel.tsx
Comment (ezeilo-su) [inline client/src/components/TabPanel/TabPanel.tsx:35]: Please let's remove unused code

File: client/src/components/TabPanel/TabPanel.tsx
Comment (ezeilo-su) [inline client/src/components/TabPanel/TabPanel.tsx:57]: This comment line is not needed here. Let's get rid of it :smiley:

File: client/src/components/TabPanel/TabPanel.tsx
Comment (ezeilo-su) [inline client/src/components/TabPanel/TabPanel.tsx:85]: Let's also get rid of this commented code

File: client/src/pages/Settings/CustomBooking/TicketList/TicketList.tsx
Comment (ezeilo-su) [inline client/src/pages/Settings/CustomBooking/TicketList/TicketList.tsx:54]: Let's remove!!!

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Good progress with this one. Please let's address those comments I left. Please also note that it's a requirement to always link the ticket being addressed in each PR. :smiley:

**Review by ezeilo-su — COMMENTED** [review]
Thank you for making the requested changes. Please do not merge until approved :smiley:

File: client/src/components/CustomBookingWraper/CustomBookingWrapper.tsx
Comment (moffatethan) [inline client/src/components/CustomBookingWraper/CustomBookingWrapper.tsx:12]: nit: use the colors library from MUI

File: client/src/components/TabPanel/TabPanel.tsx
Comment (moffatethan) [reply client/src/components/TabPanel/TabPanel.tsx:30]: Did you forget this comment Rob?

File: client/src/components/TabPanel/TabPanel.tsx
Comment (moffatethan) [inline client/src/components/TabPanel/TabPanel.tsx:44]: I don't see this code being used anywhere in this file, remove?

File: client/src/context/useAuthContext.tsx
Comment (moffatethan) [inline client/src/context/useAuthContext.tsx:11]: I prefer to not pollute this with `never` or `any`, look at your data coming back and make proper interfaces to capture different states of the profile so it excepts either of them

File: client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx
Comment (moffatethan) [inline client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx:27]: Can we import on the individual exported components collectively. 

```
import { TableRow, TableHead, . . . } from '@mui/material';
```

Similar to what you have above, just combine then. It's messy when doing different types of imports.

File: client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx
Comment (moffatethan) [inline client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx:58]: Can we move this outside of the component

File: client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx
Comment (moffatethan) [inline client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx:62]: This works for now but for integration we should use Stripe's last 4 digits they provide for the customers payment method

File: client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx
Comment (moffatethan) [inline client/src/pages/Settings/CustomBooking/CustomBookingForm/CustomBooking.tsx:88]: So, we're only ever going to have one row for the dog sitting service and nothing extra, let's just make one row here for the dog service

File: client/src/pages/Settings/CustomBooking/TicketList/TicketList.tsx
Comment (moffatethan) [inline client/src/pages/Settings/CustomBooking/TicketList/TicketList.tsx:11]: Same thing here as the previous comment 🙂

File: client/src/pages/Settings/Settings.tsx
Comment (moffatethan) [inline client/src/pages/Settings/Settings.tsx:45]: Not sure why this is on the settings page? This should be a page in the navbar for the customer.

File: client/src/pages/routes/route.ts
Comment (moffatethan) [inline client/src/pages/routes/route.ts:65]: `Custom Booking` should be `Customer Booking`

File: client/src/pages/routes/route.ts
Comment (moffatethan) [inline client/src/pages/routes/route.ts:62]: `/customer/booking`

File: client/src/pages/routes/route.ts
Comment (moffatethan) [reply client/src/pages/routes/route.ts:65]: Please update the component name too

File: client/src/pages/Settings/Settings.tsx
Comment (moffatethan) [reply client/src/pages/Settings/Settings.tsx:45]: Make sure to move the components out of the settings folder to the pages

File: server/package.json
Comment (moffatethan) [inline server/package.json:30]: What's this for?

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please bring back the sample.env - this is important for other developers on the project and potential future developers for onboarding. Shouldn't touch the file in this PR. 

Be mindful of import style too, would like to see group imports rather than individual imports when possible.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**
